larq/math.py

```diff
@@ -22,7 +22,8 @@ def sign(x):
     # Returns
         A Tensor with same type as `x`.
     """
-    return tf.sign(tf.sign(x) + 0.1)
+    ones = tf.ones_like(x)
+    return tf.where(x >= 0, ones, -ones)
```

Comment:
I'm not sure if there's any possible benefit to doing so in terms of speed/memory usage, but LCE would also correctly convert `tf.where(x >= 0, 1.0, -1.0)` if we just rely on implicit broadcasting.

Reply:
I briefly benchmarked this some time ago on a T4 GPU in [this notebook](https://colab.research.google.com/drive/16YcyqFmVNaFfXhlqItUYsoL2HF8iX8-I?usp=sharing) (although only on one input size) and it looked like explicit broadcasting using `ones_like` is faster when running with default TF kernels (especially in graph mode). If the model is XLA compiled this is optimised away and there is no performance difference. So I think using explicit broadcasting here ensures we're on the save side with respect to performance.

This is correctly supported by LCE as well, right?

Reply:
Yep it is, this will definitely work too :+1: